geps/gep-1651.md

```diff
@@ -0,0 +1,298 @@
+their defined semantics.
+
+Value | Scope
+-|-
+`Public`|The address is routable on the public internet
+`Private`|The address is routable inside a private network larger than a single cluster (ie. VPC) and MAY include RFC1918 address space
+`Cluster`|The address is routable inside the [cluster's network](https://kubernetes.io/docs/concepts/cluster-administration/networking/#how-to-implement-the-kubernetes-network-model)
+
+Values can be compared semantically - `Public` has a larger scope than `Private`, while `Private` has a larger scope than `Cluster`.
+
+### Vendor prefixed values
+
+Implementations can define custom 'routability' values by specifying a vendor prefix followed
+by a slash `/` and a custom name ie. `com.example.com/my-routability`.
+
+Comparing vendor prefixed scopes with the pre-defined ones in implementation specific.
+
+### Default Routability
+
+The default value of `routability` is implementation specific. It is RECOMMENDED that
+the default `routability` remains consistent for Gateways with the same
+`gatewayClassName`.
+
+Implementations MAY leave the `spec.infrastructure.routability` property unset and instead
+signal the default routability using the Gateway's `status.addresses`. See 'Status Addresses`
+for more details.
+
+### Mutability
+Implementations MAY prevent end-users from updating the `routability` value of a Gateway. If
+updates are allowed the semantics and behaviour will depend on the underlying implementation.
+
+### Go
+
+```go
+
+type GatewayRoutability string
+
+const (
+  GatewayRoutabilityPublic  GatewayRoutability  = "Public"
+  GatewayRoutabilityPrivate GatewayRoutability  = "Private"
+  GatewayRoutabilityCluster GatewayRoutability  = "Cluster"
+)
+
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+type GatewayInfrastructure struct {
+  // AddressScope allows the Gateway to specify the accessibility of it's addresseses
+  Routability GatewayRoutability `json:"routability"`
+}
+
+type GatewayAddress struct {
+  // Routability specifies the routable bounds of this address
+  // Predefined values are - Private, Public, Cluster
+  //
+  // This only appears on the addresses in the Gateway's status
+  Routability GatewayRoutability `json:"routability"`
+  // ...
+}
+```
+
+### YAML
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+### Semantics
+
+#### Interaction with GatewayClass
+
+An infrastructure provider may provide a pre-defined set of GatewayClasses that limit the
+routability choices of a Gateway. If the desired Gateway routability is incompatible with the
+GatewayClass it MUST set the condition `Accepted` to `False` with `Reason` set to `Invalid`.
```

Comment:
Mentioned in comments above, but we also need to describe what happens when Routability changes to an unsupported value. In any case, I think I'd prefer a more specific reason than `Invalid` for this.

Reply:
Changed to `UnsupportedRoutability`